contracts/FlowStakingCollection.cdc

```diff
@@ -0,0 +1,844 @@
+                    tokenHolderDelegatorNodeID = _tokenHolder!.getDelegatorNodeID()
+                    tokenHolderDelegatorID = _tokenHolder!.getDelegatorID()
+                }
+            }
+
+            // If the request is for a delegator, check all possible delegators for possible matches
+            if let _delegatorID = delegatorID {
+                if (tokenHolderDelegatorNodeID != nil && tokenHolderDelegatorID != nil && tokenHolderDelegatorNodeID! == nodeID && tokenHolderDelegatorID! == _delegatorID) {
+                    return true
+                }
+
+                // Look for a delegator with the specified node ID and delegator ID
+                return self.borrowDelegator(nodeID, _delegatorID) != nil 
+            } else {
+                if (tokenHolderNodeID != nil && tokenHolderNodeID! == nodeID) {
+                    return true
+                }
+
+                return self.borrowNode(nodeID) != nil
+            }
+        }
+
+        /// Function to add an existing NodeStaker object
+        pub fun addNodeObject(_ node: @FlowIDTableStaking.NodeStaker) {
+            let stakingInfo = FlowIDTableStaking.NodeInfo(nodeID: node.id)
+            let totalStaked = stakingInfo.tokensStaked + stakingInfo.tokensCommitted + stakingInfo.tokensUnstaking + stakingInfo.tokensUnstaked
+            self.unlockedTokensUsed = self.unlockedTokensUsed + totalStaked
+            self.nodeStakers[node.id] <-! node
+        }
+
+        /// Function to add an existing NodeDelegator object
+        pub fun addDelegatorObject(_ delegator: @FlowIDTableStaking.NodeDelegator) {
+            let stakingInfo = FlowIDTableStaking.DelegatorInfo(nodeID: delegator.nodeID, delegatorID: delegator.id)
+            let totalStaked = stakingInfo.tokensStaked + stakingInfo.tokensCommitted + stakingInfo.tokensUnstaking + stakingInfo.tokensUnstaked
+            self.unlockedTokensUsed = self.unlockedTokensUsed + totalStaked
+            self.nodeDelegators[delegator.nodeID] <-! delegator
+        }
+
+        /// Function to remove an existing NodeStaker object.
+        /// If the user has used any locked tokens, removing NodeStaker objects is not allowed.
+        pub fun removeNode(nodeID: String): @FlowIDTableStaking.NodeStaker? {
+            pre {
+                self.doesStakeExist(nodeID: nodeID, delegatorID: nil): "Specified node does not exist"
+                self.lockedTokensUsed == UFix64(0.0): "Cannot remove node if locked tokens are used"
+            }
+            
+            if self.nodeStakers[nodeID] != nil {
+                let stakingInfo = FlowIDTableStaking.NodeInfo(nodeID: nodeID)
+                let totalStaked = stakingInfo.tokensStaked + stakingInfo.tokensCommitted + stakingInfo.tokensUnstaking + stakingInfo.tokensUnstaked
+
+                /// Since the NodeStaker object is being removed, the total number of unlocked tokens staked to it is deducted from the counter.
```

Comment:
Only use doc comments (`///`) for declaration documentation, i.e. switch to normal comment (`//`)